main/src/selenium/actionWrapper.js

```diff
@@ -0,0 +1,224 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable no-empty-function */
+/* eslint-disable no-await-in-loop */
+const { LegacyActionSequence } = require('selenium-webdriver/lib/actions');
+const { Origin } = require('selenium-webdriver/lib/input');
+const { By } = require('selenium-webdriver/lib/by');
+
+const standardPause = 100;
+const standardMoveDuration = 200;
+const mediumPause = 300;
+const longerPause = 500;
+const longerMoveDuration = 500;
+
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+/**
+ * This class is an abstraction of the moves performed by selenium,
+ * As there may be an issue with the W3C actions implemented by selenium,
+ * there is an option to go into legacyMode
+ * In legacyMode openlayers and mapbox seem to behave differently, so there will be the need to specify
+ * the renderer
+ */
+class ActionWrapper {
+  constructor(driverForActions) {
+    this.driverForActions = driverForActions;
+  }
+  async initMap() {
+    this.map = await this.driverForActions.findElement(By.id('map'));
+  }
+  move() {
+  }
+  drag() {
+  }
+  doubleClick() {
+  }
+  async perform() {
+  }
+}
+
+const typeActionLegacySequence = 'actionLegacy';
+const typePause = 'pause';
+class LegacyActionExtended {
+  constructor(type, action) {
```

Comment:
I think this class should be split in 2, one for `Pause` and one for `Sequence`. Then each of them can implement a `run()` method that performs the correct action, and you can get rid of the `type` member and all the checks for it.